src/test/java/org/folio/dcb/controller/TransactionApiControllerTest.java

```diff
@@ -94,36 +117,68 @@ void createBorrowingPickupCirculationRequestTest() throws Exception {
+    systemUserScopedExecutionService.executeAsyncSystemUserScoped(TENANT, () -> atomicReferenceOfTransactionAuditEntityExisted.set(transactionAuditRepository.findLatestTransactionAuditEntityByDcbTransactionId(DCB_TRANSACTION_ID)
+      .orElse(null)));
+
+    TransactionAuditEntity transactionAuditEntity = atomicReferenceOfTransactionAuditEntityExisted.get();
+    Assertions.assertNotNull(transactionAuditEntity);
+    Assertions.assertNotEquals(TRANSACTION_AUDIT_DUPLICATE_ERROR_ACTION, transactionAuditEntity.getAction());
+    Assertions.assertNotEquals(DUPLICATE_ERROR_TRANSACTION_ID, transactionAuditEntity.getTransactionId());
   }
 
   @Test
   void createLendingCirculationRequestWithInvalidItemId() throws Exception {
     var dcbTransaction = createDcbTransactionByRole(LENDER);
     dcbTransaction.getItem().setId("5b95877d-86c0-4cb7-a0cd-7660b348ae5b");
 
+    String trnId = UUID.randomUUID().toString();
+
     this.mockMvc.perform(
-        post("/transactions/" + UUID.randomUUID())
+        post("/transactions/" + trnId)
           .content(asJsonString(dcbTransaction))
           .headers(defaultHeaders())
           .contentType(MediaType.APPLICATION_JSON)
           .accept(MediaType.APPLICATION_JSON))
       .andExpectAll(status().is4xxClientError(),
         jsonPath("$.errors[0].code", is("NOT_FOUND_ERROR")));
+
+    // check for transactions_audit error content.
+    AtomicReference<TransactionAuditEntity> atomicReferenceOfTransactionAuditEntityExisted = new AtomicReference<>();
+    systemUserScopedExecutionService.executeAsyncSystemUserScoped(TENANT, () -> atomicReferenceOfTransactionAuditEntityExisted.set(transactionAuditRepository.findLatestTransactionAuditEntityByDcbTransactionId(trnId)
+      .orElse(null)));
+
+    TransactionAuditEntity transactionAuditEntity = atomicReferenceOfTransactionAuditEntityExisted.get();
+    Assertions.assertNotNull(transactionAuditEntity);
+    Assertions.assertEquals(TRANSACTION_AUDIT_ERROR_ACTION, transactionAuditEntity.getAction());
   }
 
   @Test
   void createBorrowingPickupCirculationRequestWithInvalidDefaultNotExistedPatronId() throws Exception {
     var dcbTransaction = createDcbTransactionByRole(BORROWING_PICKUP);
     dcbTransaction.getPatron().setId(NOT_EXISTED_PATRON_ID);
 
+    String trnId = UUID.randomUUID().toString();
     this.mockMvc.perform(
-        post("/transactions/" + UUID.randomUUID())
+        post("/transactions/" + trnId)
           .content(asJsonString(dcbTransaction))
           .headers(defaultHeaders())
           .contentType(MediaType.APPLICATION_JSON)
           .accept(MediaType.APPLICATION_JSON))
       .andExpectAll(status().is4xxClientError(),
         jsonPath("$.errors[0].code", is("NOT_FOUND_ERROR")));
+
+    // check for transactions_audit error content.
+    AtomicReference<TransactionAuditEntity> atomicReferenceOfTransactionAuditEntityExisted = new AtomicReference<>();
```

Comment:
Instead of using AtomicReference, can we add like below. 
```
systemUserScopedExecutionService.executeAsyncSystemUserScoped(
      TENANT,
      () -> {
        TransactionAuditEntity transactionAuditEntity = transactionAuditRepository.findLatestTransactionAuditEntityByDcbTransactionId(trnId)
          .orElse(null);
        Assertions.assertNotNull(transactionAuditEntity);
        Assertions.assertEquals(TRANSACTION_AUDIT_ERROR_ACTION, transactionAuditEntity.getAction());
      });
```
I feel the above approach will make code easier to read.

Reply:
done